google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/gaxx/grpc/BigtableChannelPool.java

```diff
@@ -565,19 +583,16 @@ int getAndResetMaxOutstanding() {
      * channel has been successfully retained and it is the responsibility of the caller to release
      * it.
      */
-    private boolean retain() {
-      // register desire to start RPC
-      int currentOutstanding = outstandingRpcs.incrementAndGet();
-
-      // Rough bookkeeping
-      int prevMax = maxOutstanding.get();
-      if (currentOutstanding > prevMax) {
-        maxOutstanding.incrementAndGet();
-      }
-
+    @VisibleForTesting
+    boolean retain(boolean isStreaming) {
+      AtomicInteger counter = isStreaming ? outstandingStreamingRpcs : outstandingUnaryRpcs;
```

Comment:
I wonder if updating maxCounter should happen after the if statement? because the rpc is not actually sent on this channel so the max shouldn't be updated?